packages/flutter/test/rendering/editable_test.dart

```diff
@@ -587,4 +592,58 @@ void main() {
     editable.layout(BoxConstraints.loose(const Size(1000.0, 1000.0)));
     expect(editable.maxScrollExtent, equals(10));
   });
+
+  test('attach and detach correctly handle gesture', () {
+    GestureBindingSpy();
+    final TextSelectionDelegate delegate = FakeEditableTextState();
+    final RenderEditable editable = RenderEditable(
+      backgroundCursorColor: Colors.grey,
+      selectionColor: Colors.black,
+      textDirection: TextDirection.ltr,
+      cursorColor: Colors.red,
+      offset: ViewportOffset.zero(),
+      textSelectionDelegate: delegate,
+      text: const TextSpan(
+        text: 'test',
+        style: TextStyle(
+          height: 1.0, fontSize: 10.0, fontFamily: 'Ahem',
+        ),
+      ),
+      startHandleLayerLink: LayerLink(),
+      endHandleLayerLink: LayerLink(),
+      selection: const TextSelection(
+        baseOffset: 0,
+        extentOffset: 3,
+        affinity: TextAffinity.upstream,
+      ),
+      onSelectionChanged: (_, __, ___) { },
+    );
+    final PipelineOwner owner = PipelineOwner(onNeedVisualUpdate: () { });
+    editable.attach(owner);
+    // This should register pointer into GestureBinding.instance.pointerRouter.
+    editable.handleEvent(const PointerDownEvent(), BoxHitTestEntry(editable, const Offset(10,10)));
+    editable.detach();
+    // Make sure GestureBinding.instance.pointerRouter is removed
+
+    // ...
+  });
+}
+
+class GestureBindingSpy extends AutomatedTestWidgetsFlutterBinding {
```

Comment:
@chunhtai I'm trying to do that test but I'm quite lost :/ with this class I'm supposed to intercept the event but look like my overrides are never called. 

If I breakpoint before detach I can see the pointer has one route, and then 0 after the call. But it still unclear to me how testing that by code :/ 

Any help tips you can give me ?